src/vs/editor/contrib/format/common/formatActions.ts

```diff
@@ -235,6 +236,33 @@ export class FormatSelectionAction extends AbstractFormatAction {
 	}
 }
 
+@editorAction
+export class FormatOnPasteAction extends AbstractFormatAction {
+	constructor() {
+		super({
+			id: 'editor.action.formatOnPaste',
+			label: nls.localize('formatOnPaste.label', "Format on paste"),
+			alias: 'Format on paste',
+			precondition: EditorContextKeys.Writable
```

Comment:
We have context key which knows if for the current document there is someone that can format selections: `ModeContextKeys.hasDocumentSelectionFormattingProvider` (see [`FormatSelection`](https://github.com/Microsoft/vscode/blob/master/src/vs/editor/contrib/format/common/formatActions.ts#L219) as a sample). With that the menus will automatically update.